File: lyedu-api/src/main/java/com/lyedu/entity/CourseAttachment.java
package com.lyedu.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 课程附件实体
 *
 * @author LyEdu Team
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class CourseAttachment extends BaseEntity {

    /**
     * 课程ID
     */
    private Long courseId;

    /**
     * 附件名称
     */
    private String name;

    /**
     * 附件类型/扩展名
     */
    private String type;

    /**
     * 文件地址
     */
    private String fileUrl;

    /**
     * 排序
     */
    private Integer sort;
}

File: lyedu-admin/src/api/courseAttachment.ts
import request from '@/utils/request'

export interface AttachmentUploadResult {
  url: string
  path: string
  fileName: string
  fileSize: number
  fileType: string
  /** 是否已写入知识库，false 表示知识库表未初始化 */
  knowledgeSaved?: boolean
  /** 知识库记录 ID，用于课程附件关联 */
  knowledgeId?: number
}

export const uploadAttachment = (file: File) => {
  const form = new FormData()
  form.append('file', file)
  // 复用知识库上传接口（同 file_service），避免代理对 course-attachment/upload 的 405
  return request.post<AttachmentUploadResult>('/knowledge/upload', form, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
}

export interface CourseAttachment {
  id: number
  courseId: number
  knowledgeId: number
  name: string
  type?: string
  fileUrl: string
  sort: number
  createTime?: string
  updateTime?: string
}

export const getAttachmentsByCourseId = (courseId: number) => {
  return request.get<CourseAttachment[]>('/course-attachment', { params: { courseId } })
}

export const createAttachment = (data: {
  courseId: number
  knowledgeId: number
  sort?: number
}) => {
  return request.post('/course-attachment', data)
}

export const deleteAttachment = (id: number) => {
  return request.delete(`/course-attachment/${id}`)
}
